github-runner-manager/src/github_runner_manager/manager/runner_manager.py

```diff
@@ -242,7 +242,26 @@ def cleanup(self) -> IssuedMetricEventsStats:
         Returns:
             Stats on metrics events issued during the cleanup of runners.
         """
-        self._github.delete_runners([GitHubRunnerState.OFFLINE])
+        # RunnerManager.get_runners only get runners in the cloud provider, which can be
+        # misleading. Pending to tackle and put the logic of this function to get_runners
+        # and the in the RunnerInstance.
+
+        # We clean ACTIVE units that are offline. This is a race condition, as it can be running
+        # cloud init before starting the runner (reactive case) or runners that are restarting.
+        # We do not treat this case until better information is obtained from get_runners.
+        # get_runners returns unhealthy if the GitHub agent is not running and cloud init
+        # is running. If the unit is ACTIVE but stuck we could leave a runner forever.
+        cloud_instances_created = self.get_runners(cloud_states=[CloudRunnerState.CREATED])
+        cloud_instances_created_ids = [
+            cloud_instance.instance_id for cloud_instance in cloud_instances_created
+        ]
+        github_runners = self._github.get_runners([GitHubRunnerState.OFFLINE])
+        runners_to_delete = [
+            ghrunner
```

Comment:
`ghrunner` might not fit the naming convention for the team.
I think `gh_runner` might be better.